src/rmp/src/rmp.tcl

```diff
@@ -0,0 +1,109 @@
+############################################################################
+##
+## Copyright (c) 2019, OpenROAD
+## All rights reserved.
+##
+## BSD 3-Clause License
+##
+## Redistribution and use in source and binary forms, with or without
+## modification, are permitted provided that the following conditions are met:
+##
+## * Redistributions of source code must retain the above copyright notice, this
+##   list of conditions and the following disclaimer.
+##
+## * Redistributions in binary form must reproduce the above copyright notice,
+##   this list of conditions and the following disclaimer in the documentation
+##   and/or other materials provided with the distribution.
+##
+## * Neither the name of the copyright holder nor the names of its
+##   contributors may be used to endorse or promote products derived from
+##   this software without specific prior written permission.
+##
+## THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+## AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+## IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+## ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+## LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+## CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+## SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+## INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+## CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+## ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+## POSSIBILITY OF SUCH DAMAGE.
+##
+############################################################################
+
+# Restructuring could be done in various modes targeting area or timing.
+# 
+# Argument Description
+# liberty_file:    Liberty file with description of cells used in design. This would be passed to ABC.
+# mode:            Could be area or delay. In area mode focus is area reduction and timing may degrade. In delay mode delay would be reduced but area may increase.
+# slack_threshold: specifies slack value below which timing paths need to be analyzed for restructuring
```

Comment:
awkward.
timing path minimum slack for restructuring
what does 'need to be analyzed' mean? maybe it will  restructure it if it has less than that slack?

Reply:
Yes, all paths violating minimum slack do not qualify unless they exhibit path depth issues as well. A violating timing path with say path depth of 2 would not be re-synthesized. The path depth threshold can be controlled by an argument as well however.

Reply:
The point is not to explain it to me. It is to explain it to the sure in the documentation and this does not do an adequate job.

Reply:
Updated descrition